pkg/innerring/blocktimer.go

```diff
@@ -30,31 +30,68 @@ type (
 		cnrWrapper *container.Client // to invoke stop container estimation
 		epoch      epochState        // to specify which epoch to stop, and epoch duration
 
-		stopEstimationDMul uint32 // X: X/Y of epoch in blocks
-		stopEstimationDDiv uint32 // Y: X/Y of epoch in blocks
+		stopEstimationDMul uint32 // X: X/Y of epoch in seconds
+		stopEstimationDDiv uint32 // Y: X/Y of epoch in seconds
 
 		basicIncome subEpochEventHandler
 	}
 )
 
-func newEpochTimer(args *epochTimerArgs) *timer.BlockTimer {
-	epochTimer := timer.NewBlockTimer(
-		func() (uint32, error) {
-			return uint32(args.epoch.EpochDuration()), nil
-		},
-		func() {
-			for _, handler := range args.newEpochHandlers {
-				handler()
-			}
-		},
-	)
+type deltaHandler struct {
+	mul, div uint32
+	h        func()
+}
+
+type epochTimer struct {
+	timers []*time.Timer
+
+	eHandlers     []newEpochHandler
+	deltaHandlers []deltaHandler
+}
+
+func (et *epochTimer) reset(tickAfter, epochDuration time.Duration) {
+	et.stop()
+
+	et.timers = append(et.timers, time.AfterFunc(tickAfter, func() {
```

Comment:
The fact that it's no longer tied to blocks is something I'm still thinking about.

Reply:
... and what do you think? it is related but not on the timer layer, but somewhere above inside IR